onnxruntime/core/providers/qnn/builder/qnn_model.cc

```diff
@@ -318,6 +340,30 @@ Status QnnModel::ExecuteGraph(const Ort::KernelContext& context,
     return ORT_MAKE_STATUS(ONNXRUNTIME, FAIL, "QNN graph execute error. Error code: ", execute_status);
   }
 
+  for (const auto& qnn_input_info : qnn_input_infos_) {
+    auto ort_input_tensor = context.GetInput(qnn_input_info.ort_index);
+
+    ORT_RETURN_IF_ERROR(UnregisterQnnTensorMemory(
```

Comment:
how much of a performance hit does unregistering the mem handles on each run have?

if I understand correctly, for the mapped CPU buffer case, we are registering mem handles and then unregistering them for each run. if this is the case, I don't think it is worth caching these with `qnn_backend_manager_->GetOrRegisterContextMemHandle()`. maybe we can just register and deregister these mem handles directly.